components/admin_console/billing/billing_subscriptions/to_yearly_nudge_banner.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {useIntl, FormattedMessage} from 'react-intl';
+
+import AlertBanner from 'components/alert_banner';
+import useOpenCloudPurchaseModal from 'components/common/hooks/useOpenCloudPurchaseModal';
+import useOpenSalesLink from 'components/common/hooks/useOpenSalesLink';
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+
+import {SalesInquiryIssue} from 'selectors/cloud';
+import {getSubscriptionProduct as selectSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+
+import {AnnouncementBarTypes, CloudBanners, CloudProducts, Preferences, RecurringIntervals} from 'utils/constants';
+import {t} from 'utils/i18n';
+
+import {GlobalState} from '@mattermost/types/store';
+
+import './to_yearly_nudge_banner.scss';
+
+const ToYearlyNudgeBannerDismissable = () => {
+    const dispatch = useDispatch();
+
+    const openPurchaseModal = useOpenCloudPurchaseModal({});
+
+    const nudgeDismissed = useSelector((state: GlobalState) => getPreference(state, Preferences.CLOUD_YEARLY_NUDGE_BANNER, CloudBanners.NUDGE_TO_YEARLY_BANNER_DISMISSED)) === 'true';
+    const currentUser = useSelector(getCurrentUser);
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
+    const product = useSelector(selectSubscriptionProduct);
+    const currentProductProfessional = product?.sku === CloudProducts.PROFESSIONAL;
+    const currentProductIsMonthly = product?.recurring_interval === RecurringIntervals.MONTH;
+    const currentProductProMonthly = currentProductProfessional && currentProductIsMonthly;
+
+    const savedDismissedPref = () => {
+        dispatch(savePreferences(currentUser.id, [{
+            category: Preferences.CLOUD_YEARLY_NUDGE_BANNER,
+            name: CloudBanners.NUDGE_TO_YEARLY_BANNER_DISMISSED,
+            user_id: currentUser.id,
+            value: 'true',
+        }]));
+    };
+
+    if (nudgeDismissed) {
+        return null;
+    }
+
+    if (!isAdmin) {
+        return null;
+    }
+
+    if (!currentProductProMonthly) {
+        return null;
+    }
```

Comment:
Will this also cover the case of self-hosted ? We only want this for cloud

Reply:
But skus are the same for both cloud and self-hosted as they are both `professional`. Can we validate that the in-product banner will not show in Self-hosted? The Subscription page is not available for self-hosted so we should be good there

Reply:
Ah yes, you're right. I was confused with the actual license skus. With these we should be set 👍 Thanks for clarifying!